<launch>
	<node name="z_pid" pkg="pid" type="controller" >
		<param name="Kp" value="45" /> 
		<param name="Ki" value="0.85" />
		<param name="Kd" value="55" />
		<param name="upper_limit" value="2000" />
		<param name="lower_limit" value="-2000" />
		<param name="windup_limit" value="2000" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/global_z" />
		<param name="topic_from_plant" value="/state_z" />
		<param name="setpoint_topic" value="z_setpoint" />
		<param name="pid_enable_topic" value="pid_z_enable" />
	</node>

	<node name="y_pid" pkg="pid" type="controller" >
		<param name="Kp" value="60" /> 
		<param name="Ki" value="0" />
		<param name="Kd" value="0" />
		<param name="upper_limit" value="100" />
		<param name="lower_limit" value="-100" />
		<param name="windup_limit" value="2000" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/global_x" />
		<param name="topic_from_plant" value="/state_y" />
		<param name="setpoint_topic" value="y_setpoint" />
		<param name="pid_enable_topic" value="pid_y_enable" />
	</node>

	<node name="x_pid" pkg="pid" type="controller" >
		<param name="Kp" value="60" /> 
		<param name="Ki" value="0" />
		<param name="Kd" value="10" />
		<param name="upper_limit" value="20" />
		<param name="lower_limit" value="-20" />
		<param name="windup_limit" value="20" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/global_y" />
		<param name="topic_from_plant" value="/state_x" />
		<param name="setpoint_topic" value="x_setpoint" />
		<param name="pid_enable_topic" value="pid_x_enable" />
	</node>
	
	<node name="theta_x_pid" pkg="pid" type="controller" >
		<param name="Kp" value="-0.07" />
		<param name="Ki" value="0.0" />
		<param name="Kd" value="-0.03" />
		<param name="upper_limit" value="18" />
		<param name="lower_limit" value="-18" />
		<param name="windup_limit" value="18" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/roll" />
		<param name="topic_from_plant" value="/state_theta_x" />
		<param name="setpoint_topic" value="theta_x_setpoint_adjusted" />
		<param name="pid_enable_topic" value="pid_theta_x_enable" />
	</node>
	
	<node name="theta_y_pid" pkg="pid" type="controller" >
		<param name="Kp" value="-0.25" />
		<param name="Ki" value="0.0" />
		<param name="Kd" value="0.0" />
		<param name="upper_limit" value="18" />
		<param name="lower_limit" value="-18" />
		<param name="windup_limit" value="18" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/pitch" />
		<param name="topic_from_plant" value="/state_theta_y" />
		<param name="setpoint_topic" value="theta_y_setpoint_adjusted" />
		<param name="pid_enable_topic" value="pid_theta_y_enable" />
	</node>
	
	<node name="theta_z_pid" pkg="pid" type="controller" >
		<param name="Kp" value="-0.07" />
		<param name="Ki" value="0.0" />
		<param name="Kd" value="-0.03" />
		<param name="upper_limit" value="20" />
		<param name="lower_limit" value="-20" />
		<param name="windup_limit" value="20" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />

		<param name="topic_from_controller" value="/yaw" />
		<param name="topic_from_plant" value="/state_theta_z" />
		<param name="setpoint_topic" value="theta_z_setpoint_adjusted" />
		<param name="pid_enable_topic" value="pid_theta_z_enable" />
	</node>
</launch>
